command/pr_create.go

```diff
@@ -389,18 +396,39 @@ func determineTrackingBranch(remotes context.Remotes, headBranch string) *git.Tr
 	return nil
 }
 
-func generateCompareURL(r ghrepo.Interface, base, head, title, body string) string {
+func getPrAndIssueQueryParams(title, body string, assignees, labels, projects []string, milestone string) string {
+	queryParams := ""
+	if title != "" {
+		queryParams += "&title=" + url.QueryEscape(title)
+	}
+	if body != "" {
+		queryParams += "&body=" + url.QueryEscape(body)
+	}
+	if len(assignees) > 0 {
+		queryParams += "&assignees=" + url.QueryEscape(strings.Join(assignees, ","))
+	}
+	if len(labels) > 0 {
+		queryParams += "&labels=" + url.QueryEscape(strings.Join(labels, ","))
+	}
+	if len(projects) > 0 {
+		queryParams += "&projects=" + url.QueryEscape(strings.Join(projects, ","))
+	}
+	if milestone != "" {
+		queryParams += "&milestone=" + url.QueryEscape(milestone)
+	}
+	return strings.TrimLeft(queryParams, "&")
```

Comment:
This manual approach to query-encoding feels like it's getting out of hand. Let's use `url.Values` here and `v.Encode()` the result and have Go handle all of this for us.